Automattic/WP-Job-Manager
# Add stats section to job overlay

Fixes #2744, fixes #2745, fixes #2746

### Changes Proposed in this Pull Request

* Add stats section and daily chart to job overlay
* Add accent color, with a script to auto-detect link color by default
* Fix/tune related styles, improve job overlay on mobile
* Add a dev cli command to generate stats
* Add search impression count to the dashboard and admin tables

### Testing Instructions

* Open the [job_dashboard], and click on various jobs to view them. 
* Generate stats to show by clicking around job search opening jobs, or by the CLI helper command: `wp jm dev stats generate <postId>`
* The daily views chart goes from publish date to expiration date, make sure those are realistic. Test data generation goes from publish date to current date.
* Check on a few themes. Chart color should pick up the site's link color.

### New or Updated Hooks and Templates


* `job_manager_job_stats_chart` - Filter the data for the daily breakdown chart
* `job_manager_job_stats_summary` - Filter the sections and data for the summaries
* `job_manager_ui_accent_color` - Set an accent color manually

### Screenshot / Video

<img width="1328" alt="image" src="https://github.com/Automattic/WP-Job-Manager/assets/176949/1af66efc-f41b-40b6-9217-2cb6c11fe653">















----

| Plugin build for 78c905a09926b9af7d59f469d4b65bcaa16a02b9 <a href="#"><img width=600></a> |
| ------------------------------------------------------------ |
| 📦 [Download plugin zip](https://wpjobmanager.com/wp-content/uploads/2024/03/wp-job-manager-zip-2776-78c905a0.zip)                       |
| ▶️ [Open in playground](https://wpjobmanager.com/playground/?core=2024/03/2776-78c905a0)             |
